src/blur/meta-blur-actor.c

```diff
@@ -0,0 +1,933 @@
+        cogl_pipeline_set_layer_texture (priv->pipeline, 0, tex1);
+
+        cogl_framebuffer_draw_textured_rectangle (priv->fb, priv->pipeline, 
+                0.0f, 0.0f, priv->fb_width, priv->fb_height,
+                0.0f, 0.0f, 1.00f, 1.00f);
+
+        if (i > 0) cogl_framebuffer_finish(priv->fb);
+
+        cogl_pipeline_set_layer_texture (priv->pipeline2, 0, priv->fbTex);
+        cogl_framebuffer_draw_textured_rectangle (
+                priv->fb2, priv->pipeline2,
+                0.0f, 0.0f, priv->fb_width, priv->fb_height,
+                0.0f, 0.0f, 1.00f, 1.00f);
+
+        if (i > 0) cogl_framebuffer_finish(priv->fb2);
+    }
+}
+
+static gboolean prepare_texture(MetaBlurActor* self)
+{
+    MetaBlurActorPrivate *priv = self->priv;
+    CoglContext *ctx = clutter_backend_get_cogl_context(clutter_get_default_backend());
+    float x, y;
+    float width, height;
+    int fw, fh;
+    
+
+    if (!clutter_actor_is_visible(self)) {
+        return TRUE;
+    }
+
+    clutter_actor_get_size (self, &width, &height);
+    width = MAX(width, 1.0);
+    height = MAX(height, 1.0);
+
+    clutter_actor_get_transformed_position (self, &x, &y);
+
+    fw = cogl_framebuffer_get_width (cogl_get_draw_framebuffer ());
+    fh = cogl_framebuffer_get_height (cogl_get_draw_framebuffer ());
+
+#if 1
+    if (priv->texture == NULL) {
+        priv->texture = cogl_texture_2d_new_with_size(ctx, width, height);
+        cogl_texture_set_components(priv->texture, COGL_TEXTURE_COMPONENTS_RGBA);
+        cogl_primitive_texture_set_auto_mipmap(priv->texture, TRUE);
+
+        CoglError *error = NULL;
+        if (cogl_texture_allocate(priv->texture, &error) == FALSE) {
+            meta_warning ("cogl_texture_allocat failed: %s\n", error->message);
+            g_clear_pointer (&priv->texture, cogl_object_unref);
+            priv->gl_target = 0;
+            priv->gl_handle = 0;
+            return FALSE;
+        }
+
+        cogl_texture_get_gl_texture (priv->texture, &priv->gl_handle, &priv->gl_target);
+
+        cogl_pipeline_set_layer_texture (priv->pipeline, 0, priv->texture);
+        cogl_pipeline_set_layer_filters (priv->pipeline, 0,
+                COGL_PIPELINE_FILTER_LINEAR_MIPMAP_LINEAR,
+                COGL_PIPELINE_FILTER_LINEAR);
+    } else {
+        uint twidth, theight;
+        twidth = cogl_texture_get_width (priv->texture);
+        theight = cogl_texture_get_height (priv->texture);
+
+        int size = twidth * theight * 4;
+        uint8_t data[size];
+        memset (data, 0, size * sizeof (uint8_t));
+
+        CoglError* error = NULL;
+        cogl_texture_set_data (priv->texture, COGL_PIXEL_FORMAT_RGBA_8888_PRE, 0, data, 0, &error);
+
+        if (error) {
+            meta_warning ("clearing blur texture data failed: %s\n", error->message);
+        }
+    }
+
+    clutter_stage_ensure_current (clutter_actor_get_stage (self));
+    x = fminf(x, fw);
+    y = fminf(fh - y - height, fh);
+
+    cogl_flush ();
```

Comment:
We need to flush the Cogl drawing context so that we don't end up with garbage data being copied to the texture.